src/datachain/cache.py

```diff
@@ -55,7 +50,7 @@ def get_parsed_location(self) -> Optional[dict]:
     def get_hash(self) -> str:
         etag = f"{self.vtype}{self.location}" if self.vtype else self.etag
         return sha256(
-            f"{self.storage}/{self.parent}/{self.name}/{self.version}/{etag}".encode()
+            f"{self.storage}/{self.path}/{self.version}/{etag}".encode()
```

Comment:
Please include the actual `self.etag` to the hash and remove `vtype`. Also, the `etag` variable introduces quite a bit of confusion due to the conflict with `self.etag`.

It's better to make these changes now to avoid multiple braking changes.

Reply:
PS: `location` has to stay on the hash

Reply:
> Please include the actual self.etag to the hash and remove vtype. Also, the etag variable introduces quite a bit of confusion due to the conflict with self.etag.

Done!